# Oem7 Receiver initialization and configuration Commands, sent to Oem7 as-is.
# Refer to Oem7 Receiver documentation. 
#
# 
# Lines marked as "Do Not Remove" are needed to ensure correct operation.
# Some ROS messages are composed from a combination of Oem7 messages.
# It is critical to log individiual Oem7 message to ensure correct generation and field population.
#
# Message generation/composition as as follows:
# 
# ---------------------------------------------------------------------------------------------------
# 'GPSFix': 
#   BESTPOSB
#   BESTVELB
#   INSPVASB (on IMU-enabled units only, not needed on non-IMU units)
#   PSDROP2B (on IMU-enabled units only, not needed on non-IMU units)
#
#  Output rate: the highest rate of:  
#   BESTPOSB, BESTVELB, INSPVASB. 
# Typically, INSPVASB is logged at the highest rate and thus generates GPSFix
#
# -----------------------------------------------------------------------------------------------------
# NavSatFix: published at the same rate as GPSFix; data derived from:
#    BESTPOSB
#
# It is recommended to use GPSFix instead of NavSatFix.
#
# -----------------------------------------------------------------------------------------------------
# 'Imu':
#   INSPVASB
#   INSSTDEVB
#   CORRIMUSB or IMURATECORRIMUSB
#
#  Output rate: the highest rate of:  
#   CORRIMUSB or IMURATECORRIMUSB

# Commented out Oem7 commands are for example purposes
receiver_init_commands:
#-----------------------------------------------------------------------------------------------------
# Reset logging

- "UNLOGALL TRUE"
- "LOG NCOM1 GPGGA ONTIME 0.5"
#- "UNLOGALL THISPORT"
#- "FRESET"
#- "!PAUSE 10"

# INS Setup Commands - refer to https://docs.novatel.com/OEM7/Content/SPAN_Operation/SPAN_Operation.htm?tocpath=Operation%7CSPAN%20Operation%7C_____0
# for INS setup help.
#- "SETINSPROFILE LAND"
# IMU offsets (SETINSTRANSLATION command). If using SPAN then these must be entered precisely.
#    * ANT1 is the IMU to Primary GNSS antenna offset and MUST be entered precisely for proper INS
#       functioning. 
#    * ANT2 must be entered if using a dual antenna setup in order to take advantage of the second antenna in INS
#    * USER offsets the INS position from its default center of IMU to a specified location. To make INS positions match
#       the GNSS (BESTPOS, GPSFix etc.) outputting at antenna center then set this equal to ANT1.
#- "SETINSTRANSLATION ANT 1 <x> <y> <z> <std. x> <std. y> <std. z>
#- "SETINSTRANSLATION USER <x> <y> <z> 
# INS rotation offsets (SETINSROTATION command) 
#    * RBV (rotation from IMU body to vehcile) should be entered if the IMU is not aligned y-forward with the vehicle 
#       or calibrated even if it is for small rotational offsets
#- "SETINSROTATION RBV <x> <y> <z> <std. x> <std. y> <std. z>
#- "SETINITAZIMUTH 175 5"


#- ... etc.


#-------------------------------------------------------------------------------------------------------
# Standard Logging Requests

#----------------------------
# Status & Config
- "LOG INSCONFIGB ONCE"  # Log once to obtain initial configuration. Do not remove, needed for IMU messages.
- "LOG INSCONFIGB ONTIME 300" # Periodic update.
- "LOG RXSTATUSB ONCHANGED"

# GNSS and INS Solutions

# Do not remove these messages -------------------------------------------------------------------------- 
- "LOG BESTPOSB ONTIME 0.05" 
- "LOG BESTVELB ONTIME 0.05" 
#- "LOG BESTUTMB ONTIME 1"
- "LOG HEADING2B ONNEW"
#- "LOG PSRDOP2B ONCHANGED"
- "LOG INSPVASB ONTIME 0.01"
- "LOG INSPVAXB ONTIME 1"
- "LOG CORRIMUSB ONTIME 0.01"
- "LOG INSSTDEVB ONTIME 0.05"
#----------------------------------------------------------------------------------------------------------

- "LOG TIMEB ONTIME 1" # Can be used as a 'heartbeat' message.

# Post-Processing Messages; not directly used to generate ROS messages.
- "LOG RAWIMUSXB ONNEW"
- "LOG INSUPDATESTATUSB ONNEW"
- "LOG RAWEPHEMB ONNEW"
- "LOG GLOEPHEMERISB ONNEW"
- "LOG BDSEPHEMERISB ONNEW"
- "LOG GALFNAVEPHEMERISB ONNEW"
- "LOG GALINAVEPHEMERISB ONNEW"
- "LOG RANGEB ONTIME 1"







